### ISSUE TITLE
Neovim does not improve performance

### ISSUE THREAD

**Issue by FuzzOli87** [othree/yajs.vim#111]
Hello!

In your performance section, you suggest that people that are having issues with performance. I've been using Neovim for about a year and a half now and I have many of the same performance issues that have been reported. Slowdown when scrolling through comment sections and when cursorline is active. 

Is there something that I should know about how I configure Neovim?

**Comment by FuzzOli87** [comment]
I ran jelera/vim-javascript-syntax instead of yajs.vim and the performance bugs are gone. It was not this slow before I did an update today. Last time I ran an update was about three days ago on my linux machine. Before, I had cursorline on and scrolled up and down comments with no issues. Now it's slow to even open a JS file.

**Comment by FuzzOli87** [comment]
After fiddling with it, Everything after the 1.6 update bogs down Neovim. Not just minor scrolling but every other action. Opening Unite is slow. Doing :PlugUpdate was slow just updating the buffer. Nerdtree was super slow as well.

I tried various commits after 1.6 to see if it was something in between there. But no, the WebIDL update completely killed performance for me and made Neovim unusable. I reverted to commit for 1.6 and it works like before. 

If you need anything for debugging let me know and I'll be happy to provide.

**Comment by bwot** [comment]
2.0 - That explains why I've been having issues for a couple of days then!
Easy with vim-plug:
`Plug 'othree/yajs.vim', { 'tag': '1.6' }`

**Comment by othree** [comment]
I have a quickfix to change the web keywords to the old defs.
Could you help to see any change.

I don't have much difference between 1.6 and 2.0.
Maybe I should use option to let user switch the defs.

**Comment by FuzzOli87** [comment]
@bwot Yes, vim-plug made it pretty easy. Are you having similar slowdown? I'm not even sure why it gets so ridiculously slow.

@othree I'm not familiar with what he WebIDL does as opposed to he old defs. Just from looking at he commit, it looks like just changed to references the syntax engine uses to determine what is what in a file type. (Keyword, reserved word, etc) is this correct?

**Comment by bwot** [comment]
@FuzzOli87 Maybe not slow, but wonky in lack of a better word to describe it. I just redid my whole neovim config so not really sure what's what, but going back to 1.6 seems to have made things better.

Also had troubles in 2.0 with comments not always being greyed out if I didn't end the previous line with a semicolon (which I don't do). But that is another issue waiting to be opened.

**Comment by Starefossen** [comment]
I am having performance issues with several consecutive `//` comments like this:

```js
// Qui autem deleniti sint. Id recusandae dolorem officiis molestiae non aperiam 
// officia. Et provident exercitationem quis sit dolor. Velit eveniet nesciunt   
// omnis molestiae voluptas. Unde eaque alias eum.  Earum ut reiciendis id est.  
// Dolores placeat atque dolorem fugit. Vitae quia perferendis tempore.          
// Assumenda impedit tempore est in atque expedita. Occaecati iure cumque et.    
// Soluta magni quia sunt.  Amet quod quae veritatis aut voluptas. Animi magnam  
// temporibus quos tempore doloremque qui molestiae. Cumque nisi velit           
// consequatur praesentium. Perferendis fugiat maiores consectetur reprehenderit 
// excepturi. Dolores corporis magnam natus voluptatem praesentium deserunt      
// rerum minima. Architecto explicabo nisi iusto.                                
```

Seams to have much better performance when using the `/* */` comment block:

```js
/*
// Qui autem deleniti sint. Id recusandae dolorem officiis molestiae non aperiam 
// officia. Et provident exercitationem quis sit dolor. Velit eveniet nesciunt   
// omnis molestiae voluptas. Unde eaque alias eum.  Earum ut reiciendis id est.  
// Dolores placeat atque dolorem fugit. Vitae quia perferendis tempore.          
// Assumenda impedit tempore est in atque expedita. Occaecati iure cumque et.    
// Soluta magni quia sunt.  Amet quod quae veritatis aut voluptas. Animi magnam  
// temporibus quos tempore doloremque qui molestiae. Cumque nisi velit           
// consequatur praesentium. Perferendis fugiat maiores consectetur reprehenderit 
// excepturi. Dolores corporis magnam natus voluptatem praesentium deserunt      
// rerum minima. Architecto explicabo nisi iusto.                                
*/
```

**Comment by othree** [comment]
@Starefossen Thanks for your example
Sorry I don't feel any difference on my environment.
What is your environment?
vim or nvim?

**Comment by Starefossen** [comment]
I am on standard vim on Mac OS X. This is my `.vimrc` https://github.com/Starefossen/dotfiles/blob/master/.vimrc. I will probably be switching to nvim soon.

**Comment by chemzqm** [comment]
One problem I found is yajs have so many `fold`, since I only want to fold top level functions and class methods I removed most of them, and I get better performance.

**Comment by aminroosta** [comment]
<img width="1154" alt="screen shot 2017-05-20 at 5 25 42 pm" src="https://cloud.githubusercontent.com/assets/2920178/26275934/742bdd9e-3d81-11e7-9620-1be9a977482c.png">

I used `syntime on` and `syntime report`, a couple of regex patterns are really slowing down nvim scrolling.
- javascriptNumber `[+-]\=\%(\d\+\.\d*\|\d\+\|\.\d\+\)\%([eE][+-]\=\d\+\)\=\>`
- javascriptArrowFuncDef `[a-zA-Z_$]\k*\_s*=>`
- javascriptOpSymbols `[+\-*/%\^~=<>&|?]\+`
- javascriptBlock    `\([\^:]\s\*\)\=\zs{`
- javascriptLabel    `[a-zA-Z_$]\k*\_s*:`
- javascriptTagRef   `\<[a-zA-Z_$][0-9a-zA-Z_$]*\ze``
- javascriptString   `\z(["']\)`
- javascriptIdentifierName `\<[a-zA-Z_$][0-9a-zA-Z_$]*`

**Comment by maple-leaf** [comment]
It seems like turn on `relativenumber` will affect scrolling performance. Please check measure result on this issue https://github.com/posva/vim-vue/issues/95#issuecomment-374108025
